codepulse/src/main/scala/com/secdec/codepulse/util/ZipEntryChecker.scala

```diff
@@ -95,4 +86,33 @@ trait ZipEntryChecker {
 			stream.close
 		}
 	}
+
+	def findFirstEntry(file: File, recursive: Boolean = true)(callback: (String, ZipEntry, InputStream) => Boolean): Boolean = {
```

Comment:
Naming nitpick; `predicate` instead of `callback` since it returns a Boolean

Reply:
I'm on the fence about having a default value for the `recursive` parameter. My gut reaction was that it should be explicit, but there's nothing exactly wrong with it being default. Up to you.